### PR TITLE
feat: ecs event spec

### PR DESCRIPTION
**Pull request by olegsu**
Related to https://github.com/elastic/security-team/issues/3782
image

### CODE REVIEW

File: evaluator/opa_test.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+package evaluator
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestOpaEvaluator_decode(t *testing.T) {
```
Comment (eyalkraft) [inline evaluator/opa_test.go:27]: Nice!

File: evaluator/opa.go [hunk 2]
```diff
@@ -156,6 +159,7 @@ func (o *OpaEvaluator) decode(result interface{}) (RuleResult, error) {
 	}
 
 	err = decoder.Decode(result)
+	opaResult.Metadata.CreatedAt = now()
```
Comment (eyalkraft) [inline evaluator/opa.go:162]: Since it's only used once that could maybe be 
```suggestion
	opaResult.Metadata.CreatedAt = time.Now().UTC()
```

**Review by eyalkraft — CHANGES_REQUESTED** [review]
Good job!
small changes required

File: evaluator/opa.go (hunk 2 above)
Comment (olegsu) [reply evaluator/opa.go:162]: The reason for this was to have the ability to overwrite the value to a predictable one in the [test](https://github.com/elastic/cloudbeat/pull/346/files#diff-eb9ab193881d8642e921206e9c7c64f27a626defd0e3cc3a2f2b8e3773e46c0cR59)

File: resources/ecs/event.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	KindEvent         = "event"
+	KindMetric        = "metric"
+	KindState         = "state"
+	KindPipelineError = "pipeline_error"
+	KindSignal        = "signal"
+)
+
+// Event According to https://www.elastic.co/guide/en/ecs/current/ecs-event.html
+// event.ingested property is not part of this struct as the fleet server setting it
+type Event struct {
```
[the remark this answers is not in the document]
Comment (olegsu) [reply resources/ecs/event.go:73]: Great, I was looking this one

**Review by eyalkraft — APPROVED** [review]
Looks good!

**Comment by olegsu** [comment]
A quick update following https://github.com/elastic/cloudbeat/pull/346#discussion_r947981964

To support the `ecs` struct tags properly, we need to add to all the structs that we use `ecs` tag.  Later, implement the decoder that knows how to convert the struct into `mapstr.M` as required in the `beat.Event` (https://github.com/elastic/beats/blob/main/libbeat/beat/event.go#L44)
Similar to what is been done by [packetbeat](https://github.com/elastic/beats/blob/89bcc33a9a90cbacceaa61fceec0e8d23073574c/packetbeat/pb/event.go)

The migration to `ecs` tags can be done just after this will be merged as it mostly a refactor without any changes in terms of the structure of the outgoing event.

To close, the `Event` struct that was [removed](https://github.com/elastic/cloudbeat/pull/346#discussion_r947826187.) will be part of this PR

@eyalkraft @uri-weisman

**Comment by eyalkraft** [comment]
Thanks @olegsu! This PR looks great.
Can you please open a follow-up ticket regarding using the `ecs` tag and implementing the decoder, including the information from your last comment  and including what is the motivation for us to do so? (pros vs cons)
Thanks!

**Review by gurevichdmitry — APPROVED** [review]
@olegsu, utils.py file approved. LGTM

**Comment by olegsu** [comment]
> Thanks @olegsu! This PR looks great. Can you please open a follow-up ticket regarding using the `ecs` tag and implementing the decoder, including the information from your last comment and including what is the motivation for us to do so? (pros vs cons) Thanks!

Donehttps://github.com/orgs/elastic/projects/705/views/47?filterQuery=iteration%3A%22sprint+15%22

**Pull request merged**